src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -92,6 +92,31 @@ protected string BuildRedirectUri(string targetPath)
             return Request.Scheme + "://" + Request.Host + OriginalPathBase + targetPath;
         }
 
+        /// <summary>
+        /// Handles errors by redirecting to an error handler path if configured.  Otherwise sets the status code to 500.
+        /// </summary>
+        /// <param name="error"></param>
+        /// <returns></returns>
+        protected virtual Task HandleErrorAsync(ErrorContext error)
```

Comment:
Have you considered returning `Task<bool>` to control whether the request should continue to the next middleware (to display a nice error page via MVC) or stop immediately (to return a 302 response for instance)?